pkg/ccl/backupccl/backup_test.go

```diff
@@ -6150,68 +6147,47 @@ func TestRestoreErrorPropagates(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
+	skip.UnderStressRace(t, "multinode cluster setup times out under stressrace, likely due to resource starvation.")
+
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
+	// Fail put requests that have the word "RESTORE" in the value.
+	var RESTORE = regexp.MustCompile(`RESTORE`)
+
 	dir, dirCleanupFn := testutils.TempDir(t)
 	defer dirCleanupFn()
-	params := base.TestClusterArgs{}
-	params.ServerArgs.ExternalIODir = dir
-	// TODO(ssd): The way we inject the error requires that we
-	// intercept the actual batch request for job table write. To
-	// keep this from being flakey, we need to disable various
-	// automatic jobs.  Unfortunately, if we disable the span
-	// configuration job, we can't start a tenant.
-	params.ServerArgs.DefaultTestTenant = base.TestNeedsTightIntegrationBetweenAPIsAndTestingKnobs
-	var jobsTableKey = struct {
-		syncutil.Mutex
-		key roachpb.Key
-	}{}
-
 	var shouldFail, failures int64
-	params.ServerArgs.Knobs.Store = &kvserver.StoreTestingKnobs{
-		TestingRequestFilter: func(ctx context.Context, ba *kvpb.BatchRequest) *kvpb.Error {
-			// Intercept Put and ConditionalPut requests to the jobs table
-			// and, if shouldFail is positive, increment failures and return an
-			// injected error.
-			if !ba.IsWrite() {
-				return nil
-			}
-			jobsTableKey.Lock()
-			defer jobsTableKey.Unlock()
-			if len(jobsTableKey.key) == 0 {
-				return nil
-			}
 
-			for _, ru := range ba.Requests {
-				r := ru.GetInner()
-				switch r.(type) {
-				case *kvpb.ConditionalPutRequest, *kvpb.PutRequest:
-					key := r.Header().Key
-					if bytes.HasPrefix(key, jobsTableKey.key) && atomic.LoadInt64(&shouldFail) > 0 {
-						return kvpb.NewError(errors.Errorf("boom %d", atomic.AddInt64(&failures, 1)))
-					}
-				}
-			}
-			return nil
+	params := base.TestClusterArgs{
+		ServerArgs: base.TestServerArgs{
+			ExternalIODir: dir,
+			Knobs: base.TestingKnobs{
+				Store: &kvserver.StoreTestingKnobs{
+					TestingRequestFilter: func(ctx context.Context, ba *kvpb.BatchRequest) *kvpb.Error {
+						if atomic.LoadInt64(&shouldFail) == 0 {
+							return nil
+						}
+						// Intercept ConditionalPut requests with RESTORE in the value and,
+						// if shouldFail is positive, increment failures and return an
+						// injected error.
+						for _, ru := range ba.Requests {
+							if r, valid := ru.GetInner().(*kvpb.ConditionalPutRequest); valid && RESTORE.FindString(r.Value.String()) != "" {
+								return kvpb.NewError(errors.Errorf("boom %d", atomic.AddInt64(&failures, 1)))
+							}
+						}
+						return nil
+					},
+				},
+			},
 		},
 	}
-	params.ServerArgs.Knobs.JobsTestingKnobs = jobs.NewTestingKnobsWithShortIntervals()
```

Comment:
You mentioned that you found that the shorter intervals slows does the test. That is pretty interesting. I wonder if that is still true with your other changes here.